PostHog/PostHogContext.swift

```diff
@@ -161,4 +153,94 @@ class PostHogContext {
+                                                       name: WKApplication.didBecomeActiveNotification,
+                                                       object: nil)
+            } else {
+                NotificationCenter.default.addObserver(self,
+                                                       selector: #selector(cacheScreenSize),
+                                                       name: .init("UIApplicationDidBecomeActiveNotification"),
+                                                       object: nil)
+            }
+        #else
+            cacheScreenSize()
+        #endif
+    }
+
+    private func unregisterNotifications() {
+        #if os(iOS) || os(tvOS)
+            #if os(iOS)
+                NotificationCenter.default.removeObserver(self,
+                                                          name: UIDevice.orientationDidChangeNotification,
+                                                          object: nil)
+            #endif
+            NotificationCenter.default.removeObserver(self,
+                                                      name: UIWindow.didBecomeKeyNotification,
+                                                      object: nil)
+
+        #elseif os(macOS)
+            NotificationCenter.default.removeObserver(self,
+                                                      name: NSWindow.didBecomeKeyNotification,
+                                                      object: nil)
+            NotificationCenter.default.removeObserver(self,
+                                                      name: NSWindow.didChangeScreenNotification,
+                                                      object: nil)
+        #elseif os(watchOS)
+            if #available(watchOS 7.0, *) {
+                NotificationCenter.default.removeObserver(self,
+                                                          name: WKApplication.didBecomeActiveNotification,
+                                                          object: nil)
+            } else {
+                NotificationCenter.default.removeObserver(self,
+                                                          name: .init("UIApplicationDidBecomeActiveNotification"),
+                                                          object: nil)
+            }
+        #endif
+    }
+
+    private func updateScreenSize() {
+        screenSize = {
+            #if os(iOS) || os(tvOS)
+                return UIApplication.getCurrentWindow(filterForegrounded: false)?.bounds.size
+            #elseif os(macOS)
+                return NSApplication.shared.keyWindow?.screen?.visibleFrame.size
+            #elseif os(watchOS)
+                return WKInterfaceDevice.current().screenBounds.size
+            #else
+                return nil
+            #endif
+        }()
+    }
+
+    @objc private func cacheScreenSize() {
+        // orientation change need a nudge on next run-loop
```

Comment:
the notifications are not always delivered in the main thread already?

Reply:
yeah that would be good, also if that's not the case, check `Thread.isMainThread` and only do `DispatchQueue.main.async` if needed, I think this is also good so the value is updated as soon as possible